torchrl/envs/transforms/transforms.py

```diff
@@ -2433,3 +2433,121 @@ def __repr__(self) -> str:
+        out_keys = [f"episode_{in_key}" for in_key in in_keys]
+
+        super().__init__(in_keys=in_keys, out_keys=out_keys)
+
+    def reset(self, tensordict: TensorDictBase) -> TensorDictBase:
+        """Resets episode rewards."""
+        # Non-batched environments
+        if len(tensordict.batch_size) < 1 or tensordict.batch_size[0] == 1:
+            for out_key in self.out_keys:
+                if out_key in tensordict.keys():
+                    tensordict[out_key] = 0.0
+
+        # Batched environments
+        elif "reset_workers" in tensordict.keys():
+            for out_key in self.out_keys:
+                if out_key in tensordict.keys():
+                    tensordict[out_key][tensordict["reset_workers"]] = 0.0
+
+        return tensordict
+
+    def _call(self, tensordict: TensorDictBase) -> TensorDictBase:
+        """Updates the episode rewards with the step rewards."""
+        # Sanity checks
+        self._check_inplace()
+        for in_key in self.in_keys:
+            if in_key not in tensordict.keys():
+                return tensordict
+
+        # Update episode rewards
+        for in_key, out_key in zip(self.in_keys, self.out_keys):
+            reward = tensordict.get(in_key)
+            if out_key not in tensordict.keys():
+                tensordict.set(
+                    out_key, torch.zeros(*tensordict.shape, 1, dtype=reward.dtype)
+                )
+            tensordict[out_key] += reward
+
+        return tensordict
+
+    def transform_observation_spec(self, observation_spec: TensorSpec) -> TensorSpec:
+        """Transforms the observation spec, adding the new keys generated by RewardSum."""
+        # Retrieve parent reward spec
+        reward_spec = self.parent.specs["reward_spec"]
+
+        episode_specs = {}
+        if isinstance(reward_spec, CompositeSpec):
+
+            # If reward_spec is a CompositeSpec, all in_keys should be keys of reward_spec
+            assert all(
+                [k in reward_spec.keys() for k in self.in_keys]
+            ), "Not all in_keys are present in ´reward_spec´"
+
+            # Define episode specs for all out_keys
+            for in_key, out_key in zip(self.in_keys, self.out_keys):
+                if isinstance(reward_spec[in_key], NdUnboundedContinuousTensorSpec):
+                    episode_spec = NdUnboundedContinuousTensorSpec(
+                        shape=reward_spec.shape,
+                        device=reward_spec.device,
+                        dtype=reward_spec.dtype,
+                    )
+                else:
+                    episode_spec = UnboundedContinuousTensorSpec(
+                        device=reward_spec.device,
+                        dtype=reward_spec.dtype,
+                    )
+                episode_specs.update({out_key: episode_spec})
+
+        else:
+
+            # If reward_spec is not a CompositeSpec, the only in_key should be ´reward´
+            assert set(self.in_keys) == {
```

Comment:
We should not use asserts in any other place than tests
Here I think a KeyError would be appropriate

Reply:
ok!